### PR TITLE
Add more features to Talk - tag, star, etc

### PR DESCRIPTION
**Pull request by damms005**
The title says it all, and the file diffs too!

I think this will be an awesome addition to this extremely valuable package.

### CODE REVIEW

**Comment by damms005** [comment]
@nahid Do you mind if Travis build or equivalent is setup for this repo?

**Comment by damms005** [comment]
I erroneously proliferated features in this PR in several places, PR #135 and PR #131.

This PR consolidates the suggestions in comments in those PRs (all now closed in favour of this PR), as well as other code additions and improvements.

@nahid and @cfpinto I apologize for the delay in putting this right. I believe we are all good to merge.

Thanks

**Comment by damms005** [comment]
@nafiesl @nahid @cfpinto Please I need your opinion: because some of the additions I made adds fundamentally different behavior to Talk (e.g. the notifications feature), is it not advisable that merging this PR should necessitate bumping of Talk to the next major version?

**Comment by nafiesl** [comment]
hi @damms005 I am not sure what to review here.
> the additions I made add fundamentally different behavior to Talk

:point_up_2: are those behaviour changes already discussed with the project maintainer?

>Add more features to Talk - tag, star, etc

Regarding the PR title, I am suggesting you create different PR of each feature you contribute. So the project maintainer can easily track the changes as it needed.

**Comment by damms005** [comment]
@nafies [He's already aware of some of the changes](https://github.com/nahid/talk/pull/131#issuecomment-427471427). I simply moved everything to a single PR so it can be discussed wholistically.

> Regarding the PR title, I am suggesting you create different PR of each feature you contribute. So the project maintainer can easily track the changes as it needed.

I think it is wrong for me to call it "features" in this PR title, because it is just a single "feature": the Tag feature. This is the main feature in this PR.

So while working on this tag feature, I did some refactoring to make it easy to work with Talk. I then built on this new Tag feature to give an example of how it can be used (hence the "Star"-ing of conversation was included, system notification example, etc).

All the changes I made are all about the same single feature: Tag.

The other one is being able to add "Title" to a conversation, which was already discussed in PR#131.

**Comment by damms005** [comment]
@nafiesl @imanghafoori1 Support now added for up to Laravel v8

**Comment by damms005** [comment]
@nahid @nafiesl It's been a while this issue is opened. I hope I can convince you to merge it; else we should rather close it. I want us to agree on a tradeoff and have a midpoint to resolve this.

It was my mistake and big oversight to not have made my changes modular and opened smaller PRs, rather than this large one. I sincerely do not have the bandwidth and resources to reword this to correct this mistake.

However, Talk already has extensive and significant test coverage. If I add tests that cover the features in this PR and update the README as necessary, will this be enough to get this PR merged?

**Comment by nahid** [comment]
I'll let you know after checking
